### DISCUSSION TITLE
Meal Planner Nutrition Data

### DISCUSSION THREAD

**Discussion by gagestinson** [mealie-recipes/mealie#3743] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I am starting to count calories and macros and all that stuff. I am now using the meal planner feature for this that I have not used before. I search the Web UI as well as the issues and feature request page here on GitHub for this, but when using the meal planner, there should be a tab or option or something like that that will total up all the nutrition facts (calories, protein, fiber, carbs, etc.) for that day so that it is easier to access rathe than needing to do it manually myself.

### Please provide a concise description of the feature that would resolve your issue.

It could be an option just in the settings menu for it to be displayed or another block of data above the designated day that will show the total macro information or something. There are many ways this could be implemented I am not picky. I just think this would be a good feature.

### Please consider and list out some caveats or tradeoffs made in your design decision

Depending on the implementation it could make the page too crowded, but this might be able to be mitigated if it was a drop down option or something to be turned on or off in the setting.

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by gagestinson** [comment]
I can try to implement this feature myself but I am not a big coder at all. Not even sure what this was programmed in. Frankly my method would be skimming the code and trying something and if that doesn't work I would just ask ChatGPT how to do it lol.

**Comment by anaxmedia** [comment]
This would be super helpful for me as well. I'm using the meal planner in the same way and want to plan meals based on the number of total calories for the day.

**Comment by mfr-6** [comment]
I'm bumping this up for better visibility. This would make Mealie a perfect fit for my needs. Thank you so much for your attention and participation.

**Comment by AlexandruNegura** [comment]
Yep, this would be actually really great. This and the feature with the measurements conversion (from lb to gram, etc)

**Comment by aredon** [comment]
Fully agreed and surprised this isn't a feature on any self hosted service. Especially since what follows is the ability to generate meal plans for a day with a certain calorie or macro target. This would be immensely powerful.

**Comment by JurgenCruz** [comment]
I would also like this feature. I haven't used Python extensively or Vue, but I can give it a try and see if I can implement this. It might take me a long time though

**Comment by milannvidia** [comment]
I'm willing to give it a shot, this however would be my first time helping mealie. If done can I make a pull request or does it need an issue?

**Reply by parumpum** [reply]
I had started this a while back, and then held off for the larger upgrades going on. 

Some conflicts to take care of so the PR was closed, but I will try to get back to this and wrap it up soon unless you'd like to run with this.

https://github.com/mealie-recipes/mealie/pull/4859

**Reply by parumpum** [reply]
Got this mixed up with the filters! I'll get the filters wrapped up, then I can take a look at total nutrition data as well soon unless you want it.

**Reply by milannvidia** [reply]
for me it's the same :) i'll probably take longer

**Comment by parumpum** [comment]
Just started drafting https://github.com/mealie-recipes/mealie/pull/6579 for this and adding some meal plan filters.

**Reply by parumpum** [reply]
I also found #6399 that goes further than what I was starting with here, but doesn't include meal plan rule filters with nutrition data. I'll be setting my PR up just for the meal plan rules so the more robust nutrition information on meal plan view can move forward.

**Reply by aredon** [reply]
Worth noting too that unless I'm missing something the nutritional data is attached to the recipe but it does not specify if it's the whole batch or serving size. If you adjust the serving size mealie will not adjust the nutrition data along with the rest of the portions. So one might reasonably conclude that the nutritional data is for the whole batch - which would not be particularly useful for meal planning. So this probably needs addressed first so there's clean data?
